Source Codes/CLog/loginus.c

```diff
@@ -0,0 +1,71 @@
+#include <stdio.h>
+#include <string.h>
+#include "sqlite3.h"
+
+int main() {
+    char pass[20];
+    char user_name[20];
+
+    sqlite3 *db;
+    int rc = sqlite3_open("login.db", &db); // Open or create a SQLite database named login.db
+
+    if (rc) {
+        fprintf(stderr, "Can't open database: %s\n", sqlite3_errmsg(db)); // Error handling if the database fails to open
+        return 0;
+    }
+
+user:
+    printf("Enter Username: \n");
+    fgets(user_name, sizeof(user_name), stdin);
+    user_name[strcspn(user_name, "\n")] = '\0'; // Remove newline character if present
+
+    // Clear input buffer
+    int c;
+    while ((c = getchar()) != '\n' && c != EOF);
```

Comment:
Codacy - `Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20)`